File: 백준/Silver/11866. 요세푸스 문제 0/요세푸스 문제 0.py
import sys

input = sys.stdin.readline

n, m = list(map(int, input().strip().split()))

human_list = [i for i in range(1, n + 1)]
queue = []

while human_list:
    for i in range(m):
        human_list.append(human_list.pop(0))
    queue.append(human_list.pop())
print(f"<{', '.join(map(str, queue))}>")

File: 백준/Silver/11866. 요세푸스 문제 0/요세푸스 문제 0.js
const fs = require('fs');
const { resourceUsage } = require('process');
let input = fs.readFileSync('/dev/stdin').toString().trim().split(' ').map(Number);
const [N, K] = input;
const personList = Array.from({ length: N }, (_, i) => i + 1);
const result = [];
count = 1;
while (personList.length >= 1) {
    if (count !== K) {
        personList.push(personList.shift());
        count += 1;
    } else {
        result.push(personList.shift());
        count = 1;
    }
}

console.log(`<${result.join(', ')}>`);
